external/fv3fit/fv3fit/_shared/models.py

```diff
@@ -2,11 +2,97 @@
 import fsspec
 import yaml
 import os
+import numpy as np
 import xarray as xr
+import vcm
+
 from . import io
 from .predictor import Predictor
 
 
+@io.register("derived_model")
+class DerivedModel(Predictor):
+    _CONFIG_FILENAME = "derived_model.yaml"
+    _BASE_MODEL_DIRECTORY = "base_model_data"
+
+    def __init__(
+        self,
+        base_model: Predictor,
+        additional_input_variables: Iterable[Hashable],
+        derived_output_variables: Iterable[Hashable],
```

Comment:
Is it possible to do a sanity check on these? e.g. by checking if the derived variables are supported or by automatically getting `additional_input_variables` from the derived variables via some kind of call to `vcm.DerivedMapping`?

Reply:
The former (check if derived variables are supported) could be done by checking if they exist in `DerivedMapping._VARIABLES`, I can add this. The latter (atuomatically get `additional_input_variables`) is hard to implement with how `vcm.DerivedMapping` currently works, which is that it assumes it's the user's responsibility to initialize with an appropriate dataset and will just return KeyErrors if that isn't the case.